python/mxnet/gluon/block.py

```diff
@@ -53,18 +54,24 @@ def __init__(self, block):
 
     @staticmethod
     def create(prefix, params, hint):
-        """Creates prefix and params for new `Block`."""
+        """
+        Creates prefix, params, and profiler scope name for new `Block`.
+        The profiler scope is to support the GPU memory profiler.
+        """
         current = getattr(_BlockScope._current, "value", None)
         if current is None:
             if prefix is None:
                 if not hasattr(_name.NameManager._current, "value"):
                     _name.NameManager._current.value = _name.NameManager()
                 prefix = _name.NameManager._current.value.get(None, hint) + '_'
+            # replace the trailing underscore with colon
+            profiler_scope_name = (prefix[:-1] if prefix.endswith('_') \
```

Comment:
from the logic below, it looks like prefix can still be None, which doesn't have `.endswith`

Reply:
May I ask why? Since `prefix` is assigned a value `_name.NameManager._current.value.get(None, hint) + '_'` when it is `None`?